File: backend/src/main/java/com/stayease/service/ReviewService.java
package com.stayease.service;

import com.stayease.dto.request.ReviewRequest;
import com.stayease.dto.response.ReviewResponse;

import java.util.List;

public interface ReviewService {

    ReviewResponse addReview(ReviewRequest request, String email);

    List<ReviewResponse> getReviewsByHotel(Long hotelId);

    List<ReviewResponse> getMyReviews(String email);

    void deleteReview(Long reviewId, String email);

    Double getAverageRating(Long hotelId);
}

File: frontend/src/services/reviewService.js
import api from "./api";

export const addReview = (data) => {
    return api.post("/reviews", data);
};

export const getReviewsByHotel = (hotelId) => {
    return api.get(`/reviews/hotel/${hotelId}`);
};

export const getAverageRating = (hotelId) => {
    return api.get(`/reviews/average/${hotelId}`);
};

export const getMyReviews = () => {
    return api.get("/reviews/my");
};

export const deleteReview = (reviewId) => {
    return api.delete(`/reviews/${reviewId}`);
};